src/remapping.ts

```diff
@@ -27,14 +27,20 @@ import { SourceMapInput, SourceMapLoader } from './types';
  * it returns a falsey value, that source file is treated as an original,
  * unmodified source file.
  *
- * Pass `excludeContent` content to exclude any self-containing source file
- * content from the output sourcemap.
+ * Pass `excludeContent` to exclude any self-containing source file content
+ * from the output sourcemap.
+ *
+ * Pass `decodeMappings` to get a sourcemap with decoded mappings.
  */
 export default function remapping(
   input: SourceMapInput | SourceMapInput[],
   loader: SourceMapLoader,
-  excludeContent?: boolean
-): SourceMap {
-  const graph = buildSourceMapTree(input, loader);
-  return new SourceMap(graph.traceMappings(), !!excludeContent);
+  excludeContent?: boolean,
+  decodeMappings?: boolean,
+  segmentsAreSorted?: boolean
+): SourceMap | DecodedSourceMap {
+  const graph = buildSourceMapTree(input, loader, '', !!segmentsAreSorted);
+  return decodeMappings
+    ? graph.traceMappings()
+    : new SourceMap(graph.traceMappings(), !!excludeContent);
```

Comment:
Is seems like this should really be an option on `SourceMap`. That way the `excludeContent` flag can still take affect, and we'd return a consistent type.

Reply:
too bad this dont work
```ts
class SourceMap implements (RawSourceMap | DecodedSourceMap)
```
but we could use a function that returns `(RawSourceMap | DecodedSourceMap)`

> we'd return a consistent type

what type would that be? `RawOrDecodedSourceMap`?

```ts
export interface RawOrDecodedSourceMap extends SourceMapV3 {
  mappings: string | SourceMapSegment[][];
}
```

or copy-paste `class SourceMap` to `class DecodedSourceMap`?

Reply:
now using the `RawOrDecodedSourceMap` type, ugly but works